features/iterator-methods.yml

```diff
@@ -0,0 +1,18 @@
+name: Iterator methods
+description: The constructor and the `drop()`, `every()`, `filter()`, `find()`, `flatMap()`, `forEach()`, `from()`, `map()`, `reduce()`, `some()`, `take()` and `toArray()` methods on the `Iterator` object.
```

Comment:
Listing all the methods themselves in the description seems unnecessary and would need careful updating if anything was added/deprecated. Could probably summarize, eg "and methods for iterating, searching, processing and more" or something like that?

Reply:
I thought we listed all methods in other features previously. I guess we did this because if there is a new chunk of additions to an API in the future then they wouldn't be added to this feature but they would end up in a new feature. To then differentiate the features, the description helps by saying which additions were are talking about exactly for each of them.

Reply:
Oh if there's precedent and a plan around doing this, then great!

Maybe worth bringing up at the weekly meeting, since it sounds the root of the question is "how are features defined", and how that affects what is or is not included in the `description`.

Reply:
Other examples in which we list all of the methods in the description:

array-by-copy.yml
exp-functions.yml
set-methods.yml
trig-functions.yml

Reply:
My take on this is that we ought to name things explicitly when there's room for it. If that's not possible, name an essential "entry point" to that feature (e.g., `fetch()` rather than enumerating every part of the fetch API). If that's still impractical, then name the thing and try to summarize in the most concrete terms available to you ("The example API…").

This one is very close to being unmanageable by naming everything directly — perhaps `Iterator` and `Iterator.from()` are the entry points here? I also named the 4 most-searched-for iterator methods, according to MDN's page ranking.

```suggestion
description: The `Iterator` object is an abstract base for objects that implement the iterator protocol. It provides methods common to built-in iterators, such as `filter()`, `find()`, `map()`, and `reduce()`. You can also use the static method `Iterator.from()` to convert an existing iterable into an `Iterator`.
```

Reply:
(The [docs for `Iterator()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Iterator/Iterator#browser_compatibility) are wild, so I did not mention it or constructability. This API is aggressively hostile to being understood.)